Fix profile path and print center

ra100 opened the pull request.

ra100 committed:
Move profiles to static

Plugin was failing before slicing, because it couldn't find default
profile file. It wasn't added to egg.

donovan6000 committed:
Allows positioning sliced files. Defaults to center position if no position is supplied just like the Cura Engine plugin does

ra100 committed:
Merge pull request #1 from donovan6000/master

Allows positioning sliced files. Defaults to center position if no position is supplied just like the Cura Engine plugin does

ra100 committed:
Merge branch 'master' of github.com:Javierma/OctoPrint-Slic3r

ra100 committed:
Set threads default to number of CPUs + 1

ra100 committed:
Merge pull request #2 from ra100/ra100

Set threads default to number of CPUs + 1

ra100 committed:
Remove post_processing key

This key is not in slic3r settings

Javierma commented:
Thanks for your contribution. Sorry that I didn't say anything before. I am busy with other duties and I am also trying to see how to accept the pull request without including the new code into the pull request I did before testing it. I will try to solve it as soon as possible

Javierma commented on octoprint_slic3r/__init__.py:
```diff
@@ -166,7 +166,7 @@ def get_slicer_properties(self):
 		path = self._settings.get(["default_profile"])
 		if not path:
-			path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "profiles", "default.profile.ini")
+			path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "static/profiles", "default.profile.ini")
```
I have a doubt here, why static/profiles instead of just profiles?

ra100 replied:
Honestly, I'm not sure which way is it better. I've just did it as I've seen it in M3D-Fio. But when you ask, maybe better way will be to just let it be in profiles and add `profiles` folder here https://github.com/Javierma/OctoPrint-Slic3r/blob/master/setup.py#L27
If you think, the "profiles" folder way is better, I'll just change it in my repo and when I push changes. you'll see them right here.

Javierma replied:
Ok, I think I see what you mean. In this case, static and templates are folders used for code files, where static has *.js files and templates *.jinja2 files that generate the HTML for the user interface. Profiles folder contains settings to be used in case that there is not any profile uploaded, which would be located at .octoprint/slicingProfiles/scli3r

ra100 replied:
Well, do I get it right, that, locating profiles in profiles not in static/profiles should be better?

Javierma replied:
I think so

ra100 replied:
All is changed/fixed in later commits.

Javierma commented on octoprint_slic3r/profile.py:
```diff
@@ -181,10 +182,9 @@ class SeamPositions(object):
     support_material_contact_distance = 0.2,
     use_volumetric_e = 0,  				# 0 (unchecked) or 1 (checked)
-    threads = 2,
+    threads = min(multiprocessing.cpu_count() + 1, 16),
```
This can be a good idea to achieve a better performance, but in my opinion this parameter must be overriden during profile import (indeed I would say it will be done unless something specifies not to override this setting)

ra100 replied:
Well, I just tried, if it makes any difference with values 5 and 2 on RPi2, and it seems there is little to none performance gain, probably there is not much to do in parallel. I'll revert it to 2 threds.

Javierma replied:
Ok. Nevertheless, the user still have the option of changing the setting in Slic3r before uploading the profile so it can be equally changed.

ra100 committed:
Change gcode default values to empty strings

When no special gcodes were provided, plugin entered strings 'None' in exported
gcode.

ra100 committed:
Move profiles back to octoprint_slic3r directory

ra100 committed:
Revert threads count default to 2

ra100 committed:
Change static/profiles to profiles

Related to commit	cc1cffc

Javierma commented:
Thanks, I finally merged it in another branch
